default.nix

```diff
@@ -233,18 +233,11 @@ let overrideCabal = pkg: f: if pkg == null then null else lib.overrideCabal pkg
         done
       '';
     });
-    cabal2nixResult = src: nixpkgs.runCommand "cabal2nixResult" {
-      buildCommand = ''
-        cabal2nix file://"${src}" >"$out"
-      '';
-      buildInputs = [
-        nixpkgs.cabal2nix
-      ];
-
-      # Support unicode characters in cabal files
-      ${if !nixpkgs.stdenv.isDarwin then "LOCALE_ARCHIVE" else null} = "${nixpkgs.glibcLocales}/lib/locale/locale-archive";
-      ${if !nixpkgs.stdenv.isDarwin then "LC_ALL" else null} = "en_US.UTF-8";
-    } "";
+    cabal2nixResult = src: builtins.trace "deprecated, use haskellSrc2nix from nixpkgs" (lib.haskellSrc2nix {
```

Comment:
Maybe mention `callCabal2nix` as well since most places where `cabal2nixResult` is used would actually be a better fit for `callCabal2nix` to reduce even more boilerplate.